### ISSUE TITLE
Attack and damage DSN not showing up

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#490]
In GitLab by @gusstavoh on Aug 15, 2021, 03:04

After trying some other mods, attack and damage DSN stopped working for automated rolls (but queried rolls still work fine), both for PCs and GM. Saving throws still show up though. I've tried disabling ALL modules but DAE, DSN, libWrapper, socketlib, Tidy UI - Game Settings and Midi-QOL, and tried disabling all options and changes I've made since last time it was working (like CUB, Combat enchancers, dnd5e helpers, advanced macros, and animations modules), to no avail.
Creating a new World with the same 5 modules activated brings up no trouble and the dice work properly, with same settings on.

EDIT: after some more troubleshooting and testing, I managed to identify the source option of the problems: the option "merge Rolls to one card" is responsible for the bug. I'm checking if other mod configurations are interfering, will bring another edit if I find something.

EDIT 2: Using a new world as testing scenario with minimal mods, same problem arises. Also, the Saving Throws will only show up with the option "Save - all see Result + Rolls". If set to "all see Result + save", DSN dice also don't show up (but I guess this is intended).

**Comment by tposney** [comment]
Can you export your midi-qol settings from the misc tab so I can make sure I am testing the same setup. In any of the setups I tried the problem does not occur - so there must be a set of configuration options that I'm not seeing

**Comment by tposney** [comment]
In GitLab by @gusstavoh on Aug 17, 2021, 08:09

[fvtt-midi-qol-settings.json](https://my-gitlab-bucket.s3.amazonaws.com/640711674/76a5b4e309dc6788c52f41f09736132d5c88c7ca47d553d358d8e2f7de958770/fvtt-midi-qol-settings.json)

Here it is. I hope it helps.

**Comment by tposney** [comment]
Thanks, I've tried with your settings and it's still working for me with/without merge card set. Can you send me one of the actors that is having the problem? (Export from actor list on the right).

If that does not help I may have to connect to your world (if possible) to try and see what is going on.

**Comment by tposney** [comment]
In GitLab by @gusstavoh on Aug 17, 2021, 11:36

All actors have the same problem. Here is one of them, [fvtt-Actor-harry-yuan-ti-sorcerer.json](https://my-gitlab-bucket.s3.amazonaws.com/640711674/8bb08ffe1ac1936e53f117e266ecb2165f80cdf36fd7855ea6233d1ab08b530d/fvtt-Actor-harry-yuan-ti-sorcerer.json).

You can connect to my world at any time (though in my timezone I'm almost going to sleep, not sure if I can keep the server on). There is no pwd. #######REDACTED##########

Don't mind all the visual effects, I'm testing my powers there.

**Comment by tposney** [comment]
Do you have ipv4 address for the site?

**Comment by tposney** [comment]
In GitLab by @gusstavoh on Aug 17, 2021, 11:58

It doesn't work. My ISP blocks it somehow =(

Also: auto target ranged spells is not working for me too.

Call me on discord if you want Necrowizard#3909

**Comment by tposney** [comment]
Does the actor you exported have one of the ranged spells that is not
working?

<details><summary>...</summary>

On Tue, Aug 17, 2021 at 11:58 AM Gustavo Santos (@gusstavoh) < gitlab@mg.gitlab.com> wrote:

> Gustavo Santos <https://gitlab.com/gusstavoh> commented
> <https://gitlab.com/tposney/midi-qol/-/issues/490#note_652408687>:
>
> It doesn't work. My ISP blocks it somehow =(
>
> Also: auto target ranged spells is not working for me too.
>
> —
> Reply to this email directly or view it on GitLab
> <https://gitlab.com/tposney/midi-qol/-/issues/490#note_652408687>.
> You're receiving this email because of your account on gitlab.com. If
> you'd like to receive fewer emails, you can unsubscribe
> <https://gitlab.com/-/sent_notifications/REDACTED/unsubscribe>
> from this thread or adjust your notification settings.
>

</details>

**Comment by tposney** [comment]
In GitLab by @gusstavoh on Aug 17, 2021, 12:12

No, this one doesn't. I've made different tries with dragons frightful presence, ranged multiattack and other configurations. I'll have to check on a clean world with no modules installed to make sure though.

EDIT: tested on another world with very few modules (DAE, midi and DSN, plus requirements), still same problems.

**Comment by tposney** [comment]
Please export one of the actors for whom ranged selection is not working so I can have a look

**Comment by tposney** [comment]
In GitLab by @gusstavoh on Aug 17, 2021, 12:49

Here it is. There is a special attack called "attack everyone".
[fvtt-Actor-quillathe-_elf-ranger_.json](https://my-gitlab-bucket.s3.amazonaws.com/640711674/743cad4c79a15b9b7efd48951f1f4198e28ab6e877c4bb73bbffaf879b70dc32/fvtt-Actor-quillathe-_elf-ranger_.json)

**Comment by tposney** [comment]
Here's a screenshot of how to setup those attacks. They still need a target specification, but instead of a template shape, you specify the type of creature to attack. The range line is only relevant for special.

rwak/mwak/rsak/msak can't be ranged targets - they go via target selection prior to roll.


![Screenshot_from_2021-08-17_13-34-26](https://my-gitlab-bucket.s3.amazonaws.com/640711674/768126904356b35ef8bfffe68f1495107933a7dbda54d6fe674c2efecdb181c9/Screenshot_from_2021-08-17_13-34-26.png)

**Comment by tposney** [comment]
In GitLab by @gusstavoh on Aug 18, 2021, 02:07

Ah, I see. My greatest problem was the 30 feet creature target. May I suggest to make the usage more clear in the module's description? I thank you in any case.


About the DSN, no idea on what the problem might be?

**Comment by tposney** [comment]
I just had a thought. If you have the merge card the dice are rolled if and only if the person doing the roll has DSN enabled. So if as GM you don't roll dice then GM rolls won't show dice being rolled on other players' screens.

**Comment by tposney** [comment]
In GitLab by @gusstavoh on Aug 20, 2021, 21:22

I was doing my tests on the server only, logged as the DM and with no player connected. As a DM I have DSN enabled, so I see the dice rolling with merge cards disabled.
I'll be testing the pov from the players perspective, although so far the DSN weren't working for them either.
I'll return the results later.

EDIT: I've tested. As I've commented before, if a giant spider (for example) attacks a target, both the attack roll and the damage rolls show no DSN, but the saving throw of the target is correctly rolled with DSN. So, the problem is not the DSN completely missing, they just don't show on automated attack and/or damage rolls.
